app/models/mixins/process_tasks_mixin.rb

```diff
@@ -121,23 +118,28 @@ def invoke_api_tasks(api_client, remote_options)
         resource_ids.each do |id|
           begin
             obj = collection.find(id)
-          rescue ManageIQ::API::Client::ResourceNotFound => err
-            _log.error(err.message)
-          else
             _log.info("Invoking task #{action} on collection #{collection_name}, object #{obj.id}, with args #{post_args}")
-            begin
-              obj.send(action, post_args)
-            rescue NoMethodError => err
-              _log.error(err.message)
-            end
+            obj.send(action, post_args)
+            msg = "'#{remote_options[:task]}' successfully initiated on remote object #{obj.class}: #{id} for collection #{collection_name}"
+            task_audit_event(:success, remote_options, :message => msg)
+          rescue StandardError => err
+            msg = "'#{remote_options[:task]}' failed on remote object #{id} for collection #{collection_name}"
```

Comment:
I think the wording needs a slight change here.  We are just invoking/initiating the task, but the wording here makes it sound like the underlying request failed.  Perhaps something like:

```
"'#{remote_options[:task]}' failed to be initiated on remote object #{obj.class}: #{id} for collection #{collection_name}"
```

Reply:
Note I also assed the `#{obj.class}` because the success case has that but not the fail case.

Reply:
👍